### DISCUSSION TITLE
Announcing `@ducanh2912/next-pwa`'s migration to GitLab

### DISCUSSION THREAD

**Discussion by DuCanhGH** [DuCanhGH/next-pwa#149] [Announcements]
`@ducanh2912/next-pwa` has migrated to GitLab.

This was the result of GitHub flagging my account, organizations, and repositories as spam. GitHub has resolved the issue since then, but by the time they did that, the trust had already been broken, and the process had already completed. I, frankly, don't take this kind of issue lightly, for I fear that something like this may happen again in the future.

In the future, `@ducanh2912/next-pwa` will be developed at GitLab and mirrored to GitHub. Issues and discussions opened at GitHub will still be supported, but pull requests will be closed.

Here is [`@ducanh2912/next-pwa`'s new home](https://gitlab.com/serwist/next-pwa).

Thank you!

**Comment by FleetAdmiralJakob** [comment]
That's a fair move man. Yesterday I wanted to access serwist and your account was down. Hopefully this does not happen in the future with GitLab.

**Comment by boredland** [comment]
Sad to hear. When moving off github, why not a truly open solution like [sourcehut](https://git.sr.ht) or [codeberg](https://codeberg.org)?

**Reply by DuCanhGH** [reply]
Well, my knowledge of free Git providers only went as far as GitHub, GitLab, and Gitea, which requires you to self-host. To my defense, only these have "Git" prefixed to their names 😄 Who knows, maybe my next projects will actually be on the alternatives you mentioned. If GitLab decides to pull off some scummy move, maybe I'd, yet again, migrate Serwist and `@ducanh2912/next-pwa` to another platform.

Oh and also Cloudflare Pages seems to only support GitHub and GitLab, so there's that.

**Comment by DuCanhGH** [comment]
Hi everyone. I'd like to clarify that `@ducanh2912/next-pwa` has been moved back to GitHub. Quoting myself from https://github.com/serwist/serwist/discussions/118:

> P/s: With my compute minutes on GitLab running out much more quickly than I had anticipated, I guess I'll be back on GitHub, heh... Can't do much about being a broke bloke... Sorry for the inconvenience.

> To be honest, after a few weeks of trying out GitLab, I'd say that GitHub is still _way_ better in many ways. Managing secrets was, frankly, quite a big problem. I obviously didn't want someone to be able to extract my secret stuff through CI/CD, so I had to restrict my secrets to protected branches, meaning that other branches just had to live without Turborepo's remote caching, resulting in many wasted compute minutes. I also couldn't enforce approvals on merge requests without going Premium, so that was quite a pain. Also, Changesets doesn't officially support GitLab, so I had to fork `changesets-gitlab` (to be honest, I could just use the project, but it looked like it was in its infancy, with it still using `@actions/core` everywhere), which was quite a maintain burden (still couldn't get Releases working, heh). GitHub also doesn't support an official and publicly available way of mirroring repos, so I just had to make GitLab push to GitHub every time I pushed. So, considering everything, I guess being back on GitHub doesn't seem that bad after all. Hopefully, its bots don't suddenly decide I am worth their time again.

> ![image](https://github.com/serwist/serwist/assets/75556609/85d8d8b8-b7a8-4580-9f5a-987228b06e11)

Sorry for the inconvenience!

**Reply by FleetAdmiralJakob** [reply]
Okay, if you need help because your account gets suspended again just contact us, we will flooding their support so hopefully your account comes back pretty soon.

**Reply by DuCanhGH** [reply]
@FleetAdmiralJakob thanks for the kind words, but please refrain from doing that 😅 The GitHub Support team was really helpful and quick last time, so there's no need for such measures as far as I'm concerned.
